lib/menu.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'menu_types'
+
+class Menu
+  include MenuTypes
+
+  attr_reader :menu, :total_basket
+
+  def initialize(menu_type = MenuTypes::CHINESE)
+    @menu = menu_type
+  end
+
+  def add(food, quantity=1)
+    raise "Food type not found for #{food}" unless @menu.any? {|hsh| hsh[:food] == food }
+    @menu.each{ |elem| elem[:quantity_ordered] += quantity if elem[:food] == food }
+  end
+
+  def total
+    @menu.inject(0) {|sum, elem| sum += (elem[:quantity_ordered] * elem[:price])}
+  end
+
+  def view_basket
+    ordered_items = @menu.select {|hash| hash[:quantity_ordered] > 0 }
+    ordered_items.map {|hash| "#{hash[:food]} x#{hash[:quantity_ordered]}: £#{sprintf("%.2f", hash[:quantity_ordered]*hash[:price])}" if hash[:quantity_ordered] > 0 }.join(", ") + ", Total: £#{sprintf("%.2f", total)}"
```

Comment:
Metrics/LineLength: Line is too long. [217/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/FormatString: Favor `format` over `sprintf`. (https://github.com/bbatsov/ruby-style-guide#sprintf)